### PR TITLE
tweaks / fixes to WebUser API

### PR DESCRIPTION
**Pull request by snopoke**
## Summary
Small changes to web user API which should not affect actual API functioning:

* call `permissions.to_json()` instead of `permissions._doc`
  * functionally these are the same but `to_json` is the preferred approach
* don't set `role` directly on the WebUser object
  * setting the role is already handled correctly but the role name was also being set directly on the WebUser object
  * tests were checking the role name on the WebUser directly instead of using `get_role`

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Covered by existing tests

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by snopoke** [comment]
:eye_speech_bubble:

File: corehq/apps/api/resources/v0_5.py [hunk 1]
```diff
@@ -344,6 +344,9 @@ def get_resource_uri(self, bundle_or_obj=None, url_name='api_dispatch_detail'):
     def _update(self, bundle):
         should_save = False
         for key, value in bundle.data.items():
+            if key == "role":
+                # role handled in _set_role_for_bundle
+                continue
```
Comment (mkangia) [inline corehq/apps/api/resources/v0_5.py:349]: Can you help understand how this was even working before? I see that `obj_create` calls `_set_role_for_bundle` but this method does not. Should that be done now?

File: corehq/apps/api/tests/user_resources.py [hunk 2]
```diff
@@ -376,7 +376,7 @@ def test_create_with_preset_role(self):
                                                    content_type='application/json')
         self.assertEqual(response.status_code, 201)
         user_back = WebUser.get_by_username("test_1234")
-        self.assertEqual(user_back.role, 'Field Implementer')
+        self.assertEqual(user_back.get_role(self.domain.name).name, 'Field Implementer')
```
Comment (mkangia) [inline corehq/apps/api/tests/user_resources.py:379]: :facepalm:

File: corehq/apps/api/resources/v0_5.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/api/resources/v0_5.py:349]: That's correct. This API does not support updating a web user's role after creation.

Comment (mkangia) [reply corehq/apps/api/resources/v0_5.py:349]: thanks @snopoke

Comment (mkangia) [reply corehq/apps/api/resources/v0_5.py:349]: then is this a correct statement? `role handled in _set_role_for_bundle`

Could this be "role updates are not supported, ignore" or something else to make that clear?

**Review by mkangia — APPROVED** [review]

File: corehq/apps/api/resources/v0_5.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/api/resources/v0_5.py:349]: `_update` is called for both creating a user and updating a user so the role field here is not being ignored, it is just handled outside of this method (but only for the 'create' case). I don't know if the lack of role update for updating a user is an oversight or not but I don't want to change that in this PR.

Comment (mkangia) [reply corehq/apps/api/resources/v0_5.py:349]: that makes sense. Didn't realize this was not the actual overall "update".

**Pull request merged**